### PR TITLE
Resolved comments

### PR DESCRIPTION
**Pull request by calmbeing**
### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: core/vote/vote_manager.go [hunk 1]
```diff
@@ -34,7 +32,38 @@ type VoteManager struct {
...
+}
+
+func newVoteDataBuffer() *voteDataBuffer {
+	return &voteDataBuffer{
+		voteDataMap: make(map[uint64]*types.VoteData, maxSizeOfRecentEntry),
+		cache:       make([]uint64, 0, maxSizeOfRecentEntry),
+	}
+}
+
+func (voteDataBuffer *voteDataBuffer) add(voteData *types.VoteData) {
```
Comment (NashBC) [inline core/vote/vote_manager.go:50]: I only see one place call this add function, it is not enough, we should make sure the map is definitely the same as the vote journal, we should handle restart case also. 

As the vote map needs to sync with the vote journal, I think integrating voteDataBuffer as a part of the Vote Journal will be better. The Vote Manager no need to call this add function, we can hide this add operation within the WriteVote function of the vote journal.

File: core/vote/vote_manager.go [hunk 2]
```diff
@@ -34,7 +32,38 @@ type VoteManager struct {
...
+}
+
+type voteDataBuffer struct {
+	voteDataMap map[uint64]*types.VoteData
+	cache       []uint64
+}
+
+func newVoteDataBuffer() *voteDataBuffer {
+	return &voteDataBuffer{
+		voteDataMap: make(map[uint64]*types.VoteData, maxSizeOfRecentEntry),
```
Comment (NashBC) [inline core/vote/vote_manager.go:45]: Can we use the lru.cache, then we can manage the size easier?

Comment (NashBC) [reply core/vote/vote_manager.go:45]: Refer diffLayerCache             *lru.Cache   // Cache for the diffLayers

File: core/vote/vote_manager.go (hunk 1 above)
Comment (calmbeing) [reply core/vote/vote_manager.go]: Resolved

File: core/vote/vote_manager.go (hunk 2 above)
Comment (calmbeing) [reply core/vote/vote_manager.go]: Done

File: core/vote/vote_journal.go [hunk 3]
```diff
@@ -61,15 +91,16 @@ func (journal *VoteJournal) WriteVote(voteMessage *types.VoteEnvelope) error {
...
 	}
 
 	if lastIndex-firstIndex+1 > maxSizeOfRecentEntry {
 		if err := walLog.TruncateFront(lastIndex - maxSizeOfRecentEntry + 1); err != nil {
-			log.Warn("Failed to truncate votes journal", "err", err)
+			log.Error("Failed to truncate votes journal", "err", err)
 		}
 	}
 
+	journal.voteDataBuffer.Add(voteMessage.Data.TargetNumber, voteMessage.Data)
```
Comment (NashBC) [inline core/vote/vote_journal.go:103]: if journal.voteDataBuffer.Len() > maxSizeOfRecentEntry {
		journal.voteDataBuffer.RemoveOldest()
	}

Comment (calmbeing) [reply core/vote/vote_journal.go:103]: I didn't get your mean. LRU has the eviction mechanism, right?

**Pull request merged**